compaction/compaction.cc

```diff
@@ -268,15 +268,23 @@ struct compaction_writer {
         : compaction_writer(nullptr, std::move(writer), std::move(sst)) {}
 };
 
+struct splitting_partition_tag { };
+using splitting_partition = bool_class<splitting_partition_tag>;
```

Comment:
I think this could reduced/modernized to just  `using splitting_partition = bool_class<struct splitting_partition_tag>;

Reply:
I think github formatting tricked you here.

Reply:
`allow_split_partition` would be a better name. We are not splitting the partition here, just allowing for it.

Reply:
Actually, splitting_partition is used to denote the state that a partition is being split. compaction_descriptor::can_split_large_partition (a raw boolean) is the one that determines whether or not a partition can be split.